source/fundamentals/builders/filters.txt

```diff
@@ -11,3 +11,684 @@ Filters Builders
+      import static com.mongodb.client.model.Filters.nin;
+      import static com.mongodb.client.model.Filters.and;
+      // <MongoCollection setup code here>
+
+      Bson existsComparison = and(exists("qty"), nin("qty", 5, 8));
+      List<Document> results = new ArrayList<>();
+      collection.find(existsComparison).into(results);
+      for (Document result : results) {
+         System.out.println(result.toJson());
+      }
+
+   The Bson renders as:
+
+   .. code-block:: none
+      :copyable: false
+   
+      {
+         "qty" :
+         [
+            { "$exists" : true,
+            { "$nin" : [5, 8] }
+         ]
+      }
+
+   The output of the above code snippet should look something like this:
+   
+   .. code-block:: json
+      :copyable: false
+      
+      {"_id": 2, "color": "purple", "qty": 10, "vendor": ["C", "D"]}
+      {"_id": 4, "color": "white", "qty": 6, "vendor": ["D"]}
+      {"_id": 5, "color": "yellow", "qty": 11, "vendor": ["A", "B"]}
+      {"_id": 8, "color": "orange", "qty": 7, "vendor": ["A", "D"]}
+
+
+
+.. _evaluation:
+
+Evaluation
+----------
+
+An ``evaluation`` filter to match documemts with an expression.
+
+The evaluation operator methods include:
+
+.. list-table::
+   :header-rows: 1
+   :stub-columns: 1
+
+   * - Evaluation Operator
+     - 
+      
+   * - ``mod()``
+     - Performs a modulo operation on the value of a field and selects documents with a specified result.
+
+   * - ``regex()``
+     - Selects documents where values match a specified regular expression.
+      
+   * - ``text()``
+     - Selects documemts matching a full-text search expression.
+     
+   * - ``where()``
+     - Matches documents that satisfy a JavaScript expression.
+
+The following code snippet shows how to use an evaluation filter:
+
+.. code-block:: java
+   
+   import static com.mongodb.client.model.Filters.*;
+
+   // <MongoCollection setup code here>
+
+   collection.find(<evaluation filter>);
+
+.. example::
+
+   The following example selects documents that have a ``color`` field
+   containing the term ``green`` in the paint_purchaces collection:
+
+   .. code-block:: java
+      
+      import static com.mongodb.client.model.Filters.text;
+
+      // <MongoCollection setup code here>
+
+      Bson textComparison = text("green");
+      List<Document> results = new ArrayList<>();
+      collection.find(textComparison).into(results);
+      for (Document result : results) {
+         System.out.println(result.toJson());
+      }
+
+   The Bson renders as:
+
+   .. code-block:: none
+      :copyable: false
+   
+      { 
+         "text" : "green"
+      }
+
+   The output of the above code snippet should look something like this:
+   
+   .. code-block:: json
+      :copyable: false
+      
+      {"_id": 7, "color": "green", "qty": 8,"vendor": ["B", "C"]}
+
+   Available with MongoDB 3.2, a version 3 text index allows case-sensitive
+   and diacritic-sensitive searches.
```

Comment:
Let's omit any information about features before MongoDB server v3.6 since those are unsupported now. You could also describe what you're about to show and what a diacritic-sensitive search is.